src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/snowflake/SnowflakeColumnTypeMapper.java

```diff
@@ -57,4 +69,33 @@ public String mapToColumnType(Schema.Type kafkaType, String schemaName) {
         return "VARIANT";
     }
   }
+
+  public Schema.Type mapJsonNodeTypeToKafkaType(JsonNode value) {
+    if (value == null || value.isNull()) {
+      return STRING;
+    } else if (value.isNumber()) {
+      if (value.isShort()) {
+        return INT16;
+      } else if (value.isInt()) {
+        return INT32;
+      } else if (value.isFloat()) {
+        return FLOAT32;
+      } else if (value.isDouble()) {
+        return FLOAT64;
+      }
+      return INT64;
+    } else if (value.isTextual()) {
+      return STRING;
+    } else if (value.isBoolean()) {
+      return BOOLEAN;
+    } else if (value.isBinary()) {
+      return BYTES;
+    } else if (value.isArray()) {
+      return ARRAY;
+    } else if (value.isObject()) {
+      return STRUCT;
+    } else {
+      return null;
```

Comment:
We do not expect to hit it ever, do we? Maybe we should log at least a warning here unless we want to throw an exception and break the execution

Reply:
this is old code, just moved it. If we hit null here then we will get `throw new IllegalArgumentException("Arrays, struct and map not supported!");`
in `mapToColumnType` method